packages/flutter_tools/lib/src/commands/build_bundle.dart

```diff
@@ -101,10 +101,18 @@ class BuildBundleCommand extends BuildSubCommand {
     // Check for target platforms that are only allowed on unstable Flutter.
     switch (platform) {
       case TargetPlatform.darwin_x64:
+        if (featureFlags.isMacOSEnabled) {
```

Comment:
Aren't these all backwards?

(Follow-up: does that mean this codepath isn't tested?)

Reply:
These were wrong, I've added positive/negative test cases for both